lib/sycamore/sycamore/tests/integration/scans/test_db_scan.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from opensearchpy import OpenSearch
+
+import sycamore
+from sycamore.tests.config import TEST_DIR
+from sycamore.transforms.partition import UnstructuredPdfPartitioner
+
+
+@pytest.fixture(scope="class")
+def setup_index():
+    index_settings = {
+        "body": {
+            "settings": {
+                "index.knn": True,
+                "number_of_shards": 5,
+                "number_of_replicas": 1,
+            },
+            "mappings": {
+                "properties": {
+                    "embedding": {
+                        "type": "knn_vector",
+                        "dimension": 384,
+                        "method": {"name": "hnsw", "engine": "faiss"},
+                    },
+                    "text": {"type": "text"},
+                }
+            },
+        }
+    }
+    client = OpenSearch(**TestOpenSearchScan.OS_CLIENT_ARGS)
+
+    # Recreate before
+    client.indices.delete(TestOpenSearchScan.INDEX, ignore_unavailable=True)
+    client.indices.create(TestOpenSearchScan.INDEX, **index_settings)
+
+    yield TestOpenSearchScan.INDEX
+
+    # Delete after
+    client.indices.delete(TestOpenSearchScan.INDEX, ignore_unavailable=True)
+
+
+def filter_doc(obj, exclude):
+    return {k: v for k, v in obj.__dict__.items() if k not in exclude}
+
+
+def compare_docs(doc1, doc2):
```

Comment:
I'd suggest using a keep-list rather than an exclude list. I expect you will need to exclude more than just binary_representation. If you make https://github.com/aryn-ai/sycamore/blob/b0e3ecaf21550c3d2258dd41b5b989e23bba03a1/lib/sycamore/sycamore/writers/opensearch.py#L167 a class variable, then you could allow exactly the values the opensearch writer writes.

Reply:
Good idea - we'll eventually want to abstract this across writers too